File: app/src/main/java/closer/vlllage/com/closer/api/models/SuggestionResult.java
package closer.vlllage.com.closer.api.models;

import java.util.Date;
import java.util.List;

import closer.vlllage.com.closer.store.models.Suggestion;

public class SuggestionResult extends ModelResult {
    public List<Double> geo;
    public String name;
    public Date created;

    public static Suggestion from(SuggestionResult suggestionResult) {
            Suggestion suggestion = new Suggestion();
            suggestion.setId(suggestionResult.id);
            updateFrom(suggestion, suggestionResult);
        return suggestion;
    }

    public static Suggestion updateFrom(Suggestion suggestion, SuggestionResult suggestionResult) {
        suggestion.setName(suggestionResult.name);
        suggestion.setLatitude(suggestionResult.geo.get(0));
        suggestion.setLongitude(suggestionResult.geo.get(1));
        suggestion.setCreated(suggestionResult.created);
        return suggestion;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/SuggestionResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.store.models.Suggestion
import java.util.*

class SuggestionResult : ModelResult() {
    var geo: List<Double>? = null
    var name: String? = null

    companion object {

        fun from(suggestionResult: SuggestionResult): Suggestion {
            val suggestion = Suggestion()
            suggestion.id = suggestionResult.id
            updateFrom(suggestion, suggestionResult)
            return suggestion
        }

        fun updateFrom(suggestion: Suggestion, suggestionResult: SuggestionResult): Suggestion {
            suggestion.name = suggestionResult.name
            suggestion.latitude = suggestionResult.geo!![0]
            suggestion.longitude = suggestionResult.geo!![1]
            suggestion.created = suggestionResult.created
            return suggestion
        }
    }
}
